simba/schedule.py

```diff
@@ -218,10 +218,26 @@ def set_charging_type(self, ct, rotation_ids=None):
             if rotation_ids is None or id in rotation_ids:
                 rot.set_charging_type(ct)
 
-    def assign_vehicles(self):
+    def assign_vehicles(self, args):
+        """ Assign vehicles using the strategy given in the arguments
+        :param args: Arguments with attribute assign_strategy
+        :type args: Namespace
+        :raises NotImplementedError: if args.assign_strategy has a no allowed value
+        """
+        assign_strategy = vars(args).get("assign_strategy")
+        if assign_strategy is None or assign_strategy == "adaptive":
```

Comment:
```
assign_strategy = vars(args).get("assign_strategy", "adaptive")
if assign_strategy == "adaptive":
```
You could get cute with a dict of assign_strategy -> function, but this works as well.